Cython/Compiler/Symtab.py

```diff
@@ -931,6 +931,19 @@ def declare_cfunction(self, name, type, pos,
             var_entry.scope = entry.scope
             entry.as_variable = var_entry
         type.entry = entry
+        pos_filetype = pos[0]._file_type if pos else None
+        if (type.exception_check and type.exception_value is None and type.nogil and
+                not pos[0].is_utility_code and
+                defining and  # don't warn about external functions here - the user likely can't do anything
+                pos_filetype != 'pxd'  # again, exclude extern functions (is_pxd is unreliable here)
+            ):
+            msg = (
+                "Exception check on '%s' will always require the GIL to be acquired. Possible solutions:"
+                "\n\t1. Declare the function as 'noexcept' if you're sure the function should not raise exceptions."
+             ) % name
+            if type.return_type.is_void:
+                msg += "\n\t2. Return a dummy integer for Cython to use as an exception flag."
+            performance_hint(pos, msg)
```

Comment:
Given that you managed to write more or less the same message twice in two entirely different wordings but with just a few additional conditions in the second incarnation, this loudly screams for a shared helper function.

Reply:
Also, "requires to be reacquired" sounds funny. Maybe there's a different word that you could use for one or the other.